src/integrations/editor/DiffViewProvider.ts

```diff
@@ -79,6 +79,15 @@ export class DiffViewProvider {
 		if (!this.relPath || !this.activeLineController || !this.fadedOverlayController) {
 			throw new Error("Required values not set")
 		}
+
+		// --- Fix to prevent duplicate BOM ---
+		// Strip potential BOM from incoming content. VS Code's `applyEdit` might implicitly handle the BOM
+		// when replacing from the start (0,0), and we want to avoid duplication.
+		// Final BOM is handled in `saveChanges`.
+		if (accumulatedContent.startsWith("\ufeff")) {
+			accumulatedContent = accumulatedContent.slice(1) // Remove the BOM character
+		}
```

Comment:
Looking at the logic here, I feel this condition is incorrect.  This will remove the `\ufeff` character from a file that ONLY had 1 at the beginning.
```suggestion
		if (accumulatedContent.startsWith("\ufeff\ufeff")) {
			accumulatedContent = accumulatedContent.slice(1) // Remove the BOM character
		}
```

Reply:
This also is a pragmatic fix to an agentic problem.  While this will correct the output, we should try to fix the underlying problem with what is causing the model to provide this in the first place.

Reply:
@dcbartlett 

> Looking at the logic here, I feel this condition is incorrect. This will remove the \ufeff character from a file that ONLY had 1 at the beginning.

No, as I mentioned in the comment, VS Code's `applyEdit` might implicitly handle the BOM. This means VS Code is aware of the original file encoding and preserves the BOM if the file is UTF-8 with BOM. Therefore, when using `vscode.workspace.applyEdit(edit)`, the edit is expected to contain no BOM in the input.


> This also is a pragmatic fix to an agentic problem. While this will correct the output, we should try to fix the underlying problem with what is causing the model to provide this in the first place.

Makes sense. I think the real issue is a mismatch between the read and write APIs. When reading the file content, it uses `fs.readFile` with UTF-8 encoding, but when writing, it uses `vscode.workspace.applyEdit`. Somehow, `fs.readFile` treats the BOM as part of the text, which is why the BOM ends up in the content.

This also fixes the problem:

```
			// this.originalContent = await fs.readFile(absolutePath, "utf-8")
			const fileUri = vscode.Uri.file(absolutePath);
			const fileContentUint8Array = await vscode.workspace.fs.readFile(fileUri);
			const decoder = new TextDecoder('utf-8');  // handles BOM correctly
			this.originalContent = decoder.decode(fileContentUint8Array);
```

You can see it at this line:  
https://github.com/cline/cline/blob/6e40ec16f3d2cfb173fbf45c2339a0b69cfa487a/src/integrations/editor/DiffViewProvider.ts#L46

But I didn't want to change only this one among many uses of `fs.readFile`. So I placed my PR close to where the issue actually happens (inside of `update` func).